raulin0/backup-juggler
# ci: add CI pipeline configuration

# Description
This pull request adds the CI pipeline configuration to enable continuous integration for the project. The changes include the addition of a GitHub Actions workflow file (pipeline.yml) and necessary steps to ensure the project builds successfully, runs tests, and generates coverage reports.


# Changes Made
- Added pipeline.yml GitHub Actions workflow file.
- Added steps to copy the files from the repository, install Python, install project dependencies using Poetry, run tests, and upload coverage reports to Codecov.


# Motivation and Context
The introduction of a CI pipeline ensures that every code change undergoes automated testing and validation. This improves code quality, detects issues early, and helps maintain project stability. By automating the build and test process, developers can focus on writing code while having confidence in the project's integrity.


# How to Test
1. Make sure the changes from this pull request are merged into the target branch.
2. Push or create a pull request with new code changes.
3. Check the Actions tab in the GitHub repository to ensure the CI pipeline is triggered.
4. Monitor the pipeline's progress and verify that each step completes successfully.
5. Inspect the generated coverage reports to ensure they accurately reflect the test coverage.
6. Validate that the coverage reports are uploaded to Codecov and can be accessed.


# Checklist
 - [x] Verified that the CI pipeline runs successfully.
  - [x] Tested the build and test process locally before submitting the pull request.
  - [x] Ensured that the coverage reports are generated correctly.
  - [x] Reviewed and addressed any warnings or issues reported by the CI pipeline.
  - [x] Updated the project documentation to reflect the addition of the CI pipeline.
  - [x] Tested the changes against relevant use cases and scenarios.
  - [x] Notified the team about the pull request and requested a thorough review.
